src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertFrom-SddlString.cs

```diff
@@ -0,0 +1,187 @@
+
+            PSObject result = new PSObject();
+
+            result.Properties.Add(new PSNoteProperty("Owner", owner));
+            result.Properties.Add(new PSNoteProperty("Group", group));
+            result.Properties.Add(new PSNoteProperty("DiscretionaryAcl", discretionaryAcl));
+            result.Properties.Add(new PSNoteProperty("SystemAcl", systemAcl));
+
+            WriteObject(result);
+        }
+
+        /// <summary>
+        /// Translates a SID into a NT Account
+        /// </summary>
+        private string ConvertToNtAccount(SecurityIdentifier sid)
+        {
+            try
+            {
+                return sid.Translate(typeof(NTAccount)).ToString();
+            }
+            catch {}
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the access rights that apply to an access mask, preferring right types
+        /// of 'Type' if specified.
+        /// </summary>
+        private List<string> GetAccessRights(long accessMask, string type)
+        {
+            IDictionary<string, Type> rightTypes = new Dictionary<string, Type>
+            {
+                { "FileSystemRights", typeof(System.Security.AccessControl.FileSystemRights) },
+                { "RegistryRights", typeof(System.Security.AccessControl.RegistryRights) },
+#if !CORECLR
+                { "ActiveDirectoryRights", typeof(System.DirectoryServices.ActiveDirectoryRights) },
+#endif
+                { "MutexRights", typeof(System.Security.AccessControl.MutexRights) },
+                { "SemaphoreRights", typeof(System.Security.AccessControl.SemaphoreRights) },
+#if !CORECLR
+                { "CryptoKeyRights", typeof(System.Security.AccessControl.CryptoKeyRights) },
+#endif
+                { "EventWaitHandleRights", typeof(System.Security.AccessControl.EventWaitHandleRights) }
+            };
+
+            List<Type> typesToExamine = rightTypes.Values.ToList();
+            
+            // If they know the access mask represents a certain type, prefer its names
+            // (i.e.: CreateLink for the registry over CreateDirectories for the filesystem)
+            if (!String.IsNullOrEmpty(type))
+            {
+                typesToExamine.Insert(0, rightTypes[type]);
+            }
+            
+            // Stores the access types we've found that apply
+            List<string> foundAccess = new List<string>();
+
+            // Store the access types we've already seen, so that we don't report access
+            // flags that are essentially duplicate. Many of the access values in the different
+            // enumerations have the same value but with different names.
+            HashSet<long> foundValues = new HashSet<long>();
+
+            // Go through the entries in the different right types, and see if they apply to the
+            // provided access mask. If they do, then add that to the result.
+            foreach (var rightType in typesToExamine)
```

Comment:
@iSazonov I also have doubts about current method of converting ACE AccessMask into access flag enum based strings. There are situations when flag cannot be converted without additional logic i.e. [FileSystemRights](https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem.AccessControl/ref/System.IO.FileSystem.AccessControl.cs#L83&L85) contains entries with duplicate flag values - some applies only for files, other only for containers. I guess that this won't be solved by requiring Type parameter.